lib/stdlib/src/erl_eval.erl

```diff
@@ -503,6 +511,28 @@ find_maxline(LC) ->
     _ = erl_parse:map_anno(F, LC),
     erase('$erl_eval_max_line').
 
+fun_used_bindings(Fun, Cs, Bs, Fuv) ->
+    {Used,InnerFuv} =
+       case Fuv of
+          %% If this clause is in our fun used vars tree,
+          %% then we do not need to compute to traverse it again.
+           #{Cs := UsedAndFuv} ->
+               UsedAndFuv;
+
+           #{} ->
+                %% Save only used variables in the function environment.
+                %% {value,L,V} are hidden while lint finds used variables.
+                {Fun1, _} = hide_calls(Fun, 0),
```

Comment:
Because we still have to hide calls, if the `Fun` has `{value, _, _}` in it, we won't be able to lookup its clauses. If we merge #5493, then we will improve the "cache hit".